server/src/main/java/com/thoughtworks/go/server/web/AuthorizationInterceptor.java

```diff
@@ -76,6 +78,18 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
         return true;
     }
 
+    private String getValueFromRequest(HttpServletRequest request, String parameter) {
+        String parameterValue = request.getParameter(parameter);
+        if (parameterValue != null) {
+            return parameterValue;
+        }
```

Comment:
This is problematic: If a user foo has access to `pipelineA`, but not `pipelineB` — (s)he can possibly craft this url for the pipeline history page and get access to pipelineB, because the impl seems to prefer using the query parameter, and ignore the path param.

```
GET /go/tab/pipeline/history/pipelineB?pipelineName=pipelineA
```